http-server/src/test/java/io/airlift/http/server/TestHttpServerProvider.java

```diff
@@ -392,17 +394,27 @@ public void testShowStackTraceDisabled()
     public void testStop()
             throws Exception
     {
-        createAndStartServer();
-
+        CountDownLatch latch = new CountDownLatch(1);
+        DummyServlet servlet = new DummyServlet(Optional.of(latch));
+        createAndStartServer(servlet);
         try (HttpClient client = new JettyHttpClient()) {
-            URI uri = URI.create(httpServerInfo.getHttpUri().toASCIIString() + "/?sleep=50000");
+            URI uri = URI.create(httpServerInfo.getHttpUri().toASCIIString());
             Request request = prepareGet().setUri(uri).build();
             HttpResponseFuture<?> future = client.executeAsync(request, createStatusResponseHandler());
 
+            // wait until the request is active
+            while (servlet.getRequestsReceived() == 0) {
+                Thread.sleep(100);
+            }
+
+            // call stop when the request is still active (waiting on the latch)
             server.stop();
 
+            // let the request proceed
+            latch.countDown();
```

Comment:
Maybe put a `sleep(100)` before this so the stop has a chance to really kick in (since the server is multithreaded)